test/parse.test.ts

```diff
@@ -195,6 +196,81 @@ See more help with --help`)
       })
     })
 
+    describe('args - no args passed in, with defaults', () => {
+      it('two args: only first is required, only second has a default', () => {
+        expect(() => {
+          parse([], {
+            args: [
+              {name: 'arg1', required: true},
+              {name: 'arg2', required: false, default: 'some_default'},
+            ],
+          })
+        }).to.throw(`Missing 1 required arg:
+arg1
+See more help with --help`)
+      })
+
+      it('two args: only first is required, only first has a default', () => {
+        expect(() => {
+          parse([], {
+            args: [
+              {name: 'arg1', required: true, default: 'my_default'},
+              {name: 'arg2', required: false},
+            ],
+          })
+        }).to.not.throw()
+      })
+
+      it('two args: both have a default, only first is required', () => {
+        expect(() => {
+          parse([], {
+            args: [
+              {name: 'arg1', required: true, default: 'my_default'},
+              {name: 'arg2', required: false, default: 'some_default'},
+            ],
+          })
+        }).to.not.throw()
+      })
+
+      // these set of permutations are not practical (first arg not required, but second is)
```

Comment:
please note this. I suppose if the dev doing this wants to shoot themselves in the foot, they can... (not sure if we want to handle this special case)

Reply:
can we error out in this case? A command should always have all optional followed by all required arguments

Reply:
will do!